kedro-datasets/kedro_datasets/spark/spark_dataset.py

```diff
@@ -286,28 +277,36 @@ def __init__(  # pylint: disable=too-many-arguments
                 if ``filepath`` prefix is ``hdfs://``. Ignored otherwise.
         """
         credentials = deepcopy(credentials) or {}
+        # protocol, _ = get_protocol_and_path(filepath)
+        # if protocol == "hdfs":
+        #     # adhoc fix for hdfs on spark
+        #     filepath = _split_filepath(filepath)[1]
+        # else:
+        #     filepath = get_protocol_and_path(filepath)[1]
         fs_prefix, filepath = _split_filepath(filepath)
+        path = PurePosixPath(filepath)
         exists_function = None
         glob_function = None
 
-        if fs_prefix in ("s3a://", "s3n://"):
-            if fs_prefix == "s3n://":
-                warn(
-                    "'s3n' filesystem has now been deprecated by Spark, "
-                    "please consider switching to 's3a'",
-                    DeprecationWarning,
-                )
+        if not filepath.startswith("/dbfs") and _deployed_on_databricks():
+            logger.warning(
```

Comment:
Outside of the scope of your PR, just a note to myself that this (and the accompanying test) should be changed to use `warn(...)`.